app/main.py

```diff
@@ -1,8 +1,21 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people.update({self.name: self})
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    for person in people:
+        Person(person["name"], person["age"])
+
+    for person in people:
+        new_person = Person.people[person["name"]]
+        if "wife" in person and person["wife"]:
+            new_person.wife = Person.people.get(person["wife"])
+        if "husband" in person and person["husband"]:
+            new_person.husband = Person.people.get(person["husband"])
```

Comment:
The code assumes that if a person has a 'wife' or 'husband' key, the corresponding person is already created and present in `Person.people`. Ensure that the input list `people` is ordered such that all referenced spouses are created before they are assigned. Otherwise, this could lead to `None` being assigned if the spouse is not yet created.